torchdata/datapipes/iter/util/dataframemaker.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+from functools import partial
+from typing import List, Optional, TypeVar
+
+from torchdata.datapipes import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe
+
+try:
+    import torcharrow
+except ImportError:
+    torcharrow = None
+
+T_co = TypeVar("T_co")
+
+
+@functional_datapipe("convert_to_dataframe")
+class DataFrameMakerIterDataPipe(IterDataPipe):  # IterDataPipe[torcharrow.IDataFrame[T_co]]
```

Comment:
Open to suggestion for a better name

Reply:
For functional API, can this be `dataframe` to simulate the `torch.numpy`?